### ISSUE TITLE
Rule Change: Allow excluding specific function calls from no-magic-numbers rule

### ISSUE THREAD

**Issue by electriquo** [eslint/eslint#19956]
### What rule do you want to change?

no-magic-numbers

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new option

### Example code

```js
test('some test', () => {
  expect.assertions(1); // should not trigger no-magic-numbers
  const result = someFunction(42); // should trigger no-magic-numbers
  expect(true).toBeTruthy();
});
```

### What does the rule currently do for this code?

The error `No magic number: 1` is thrown for the given snippet

### What will the rule do after it's changed?

Introduce a new `ignoreFunctions` option to the `no-magic-numbers` rule to allow ignoring specific function calls, such as:

```js
'no-magic-numbers': ['error', {
  ignoreFunctions: [
    { object: 'expect', method: 'assertions' }
  ]
}]
```

Or some equivalent way to declare function calls where numeric arguments are accepted and should not be considered "magic".

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @electriquo, thanks for the issue!

As stated in the [docs](https://eslint.org/docs/latest/rules/no-magic-numbers), this rule is currently [frozen](https://eslint.org/docs/latest/contribute/core-rules#frozen-rules) and is not accepting feature requests. You can always copy the rule source code into your custom rule and modify it to fit your needs.

**Comment by electriquo** [comment]
@mdjermanovic Thank you for the quick response. Could you please share how would you recommend to bypass the issue rather than implementing the new option?

**Comment by mdjermanovic** [comment]
I think the only way is to make a custom rule and implement the new option as there doesn't seem to be any workaround that would allow only specific functions calls (or any function calls) with the current configuration options.

**Comment by electriquo** [comment]
@mdjermanovic Thank you for the assistance and I still wish that it will be unfrozen :)
